pages/merchant/[username].tsx

```diff
@@ -0,0 +1,86 @@
+import { useRouter } from "next/router"
+import React from "react"
+import ButtonGroup from "react-bootstrap/ButtonGroup"
+import Container from "react-bootstrap/Container"
+import Image from "react-bootstrap/Image"
+
+import { useQuery } from "@apollo/client"
+
+import { RECIPIENT_WALLET_ID } from "../../lib/graphql/query"
+import styles from "./_user.module.css"
+import reducer, { ACTIONS } from "./_reducer"
+import ParsePayment from "../../components/ParsePOSPayment"
+
+function ReceivePayment() {
+  const router = useRouter()
+  const { username } = router.query
+
+  const { error: usernameError, data } = useQuery(RECIPIENT_WALLET_ID, {
+    variables: {
+      username,
+    },
+  })
+
+  const [state, dispatch] = React.useReducer(reducer, {
+    currentAmount: "",
+    createInvoice: false,
+    walletCurrency: data?.accountDefaultWallet.walletCurrency || "USD",
+    username: username,
+    paymentStatus: "",
+  })
+
+  React.useEffect(() => {
+    if (state.walletCurrency === data?.accountDefaultWallet.walletCurrency) return
+    dispatch({
+      type: ACTIONS.UPDATE_WALLET_CURRENCY,
+      payload: data?.accountDefaultWallet.walletCurrency,
+    })
+    dispatch({ type: ACTIONS.UPDATE_USERNAME, payload: username })
+  }, [data?.accountDefaultWallet.walletCurrency, state, username, data])
```

Comment:
It is unnecessary to depend on ```data?.accountDefaultWallet.walletCurrency``` and ```data```, instead you can just depend on ```data``` since objects are expected to be immutable so if ```data?.accountDefaultWallet.walletCurrency``` changes, so will ```data```.

Reply:
Yeah. That was an oversight. Thanks for pointing it out👍